bgp/gobgp.go

```diff
@@ -62,15 +63,55 @@ type bgpDeclTask struct {
 	listenPort    uint16
 
 	bgpStarted bool
+
+	yclient *ygnmi.Client
+
+	appliedStateMu       sync.Mutex
+	appliedState         *oc.Root
+	appliedBGP           *oc.NetworkInstance_Protocol_Bgp
+	appliedRoutingPolicy *oc.RoutingPolicy
 }
 
 // newBgpDeclTask creates a new bgpDeclTask.
 func newBgpDeclTask(zapiURL string, listenPort uint16) *bgpDeclTask {
+	appliedState := &oc.Root{}
+	// appliedBGP is the SoT for BGP applied configuration. It is maintained locally by the task.
+	appliedBGP := appliedState.GetOrCreateNetworkInstance(fakedevice.DefaultNetworkInstance).GetOrCreateProtocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, fakedevice.BGPRoutingProtocol).GetOrCreateBgp()
+	appliedBGP.PopulateDefaults()
+	appliedRoutingPolicy := appliedState.GetOrCreateRoutingPolicy()
+	appliedRoutingPolicy.PopulateDefaults()
+
 	return &bgpDeclTask{
 		zapiURL:    zapiURL,
 		bgpServer:  server.NewBgpServer(),
 		listenPort: listenPort,
+
+		appliedState:         appliedState,
+		appliedBGP:           appliedBGP,
+		appliedRoutingPolicy: appliedRoutingPolicy,
+	}
+}
+
+func updateAppliedStateHelper[T ygot.GoStruct](yclient *ygnmi.Client, path ygnmi.SingletonQuery[T], appliedState T) {
```

Comment:
optional: inline this into updateAppliedState  so that someone doesn't call it instead of updateAppliedState. it's only 3 lines

Reply:
I'm still just very slightly for keeping this since it might be applicable for other handlers